### ISSUE TITLE
Simple functions in GTX now need to define GLM_ENABLE_EXPERIMENTAL to use

### ISSUE THREAD

**Issue by PlutoniumHeart** [g-truc/glm#732]
Hi,

I am using the 0.9.9 version of glm lib, and I was using the API glm::angle() provided by gtx extension, calculation of the angle between 2 vectors seems simple enough, is there a reason why it is still in the experimental APIs? or is there an alternative functions that is from gtc extension?

Thanks!

**Comment by krux02** [comment]
```
dot(a,b) = |a| * |b| * cos(α)
dot(a,b) / (|a| * |b|) = cos(α)
α = acos(dot(a,b) / (|a| * |b|) = cos(α)) | α < π
α = acos(dot(a / |a|, b / |b|) = cos(α)) | α < π
```

so when you have normalized vectors, this can be reduced to `acos(dot(a,b))`.  But supported by default should only be what is also available by default in glsl, because glm is as close to glsl as possible. So when glsl has `angle`, then glm should also have it.

**Comment by acaudwell** [comment]
Similarly I have code in various places using length2() (yes I could change this to dot(a,a)) so needed to now define GLM_ENABLE_EXPERIMENTAL as the function is in gtx/norm.hpp.

I'm a bit confused about this new requirement to define GLM_ENABLE_EXPERIMENTAL given the fact that it's experimental is already made clear by the gtx/ in the path to the header.

**Comment by Groovounet** [comment]
Hi,

Basically, I am trying to discourage using GTX extensions as typically there are not well tested if there is any test.

Maybe, this is the opportunity to collect what functionnally in there is really important for the community and remove it from experimental.

Thanks,
Christophe

**Comment by tgurr** [comment]
> opportunity to collect what functionnally in there is really important for the community

libetonyek (https://wiki.documentfoundation.org/DLP/Libraries/libetonyek) required for LibreOffice needs `glm/gtx/io.hpp` and thus the current version 0.1.8 fails to build with glm-0.9.9.0:

```
[...]
checking for glm/glm.hpp... yes
checking for glm/gtx/io.hpp... no
configure: error: Required glm headers not found.
[...]
```

**Comment by Lord-Kamina** [comment]
I experienced the same issue updating to glm 0.9.9.0; we're using diagonal4x4.

**Comment by Groovounet** [comment]
This issue should be fixed in master branch for GLM 0.9.9.4 release.

Thanks!
